File: Solution with python/November LeetCoding Challenge 2021/1286. Iterator for Combination.py
v = []
class CombinationIterator
    
    def __init__(self, s str, lent int)
        v.clear()
        l = len(s)
        
        for mask in range ( 1, 1  l ) 
            ret = 
            cnt = 0
            
            for i in range ( 0, l ) 
                if ( ( 1  i ) & mask ) 
                    ret += s[i]
                    cnt = cnt + 1
                    
            if ( cnt == lent ) 
                v.append ( ret )
                
        
        v.sort()
        v.reverse()
        

    def next(self) - str
        ret = v[-1]
        v.pop()
        return ret

    def hasNext(self) - bool
        return ( len(v)  0 )


# Your CombinationIterator object will be instantiated and called as such
# obj = CombinationIterator(characters, combinationLength)
# param_1 = obj.next()
# param_2 = obj.hasNext()

File: November LeetCoding Challenge 2021/1286. Iterator for Combination.cpp
class CombinationIterator {
public:
    vector<string> v;
    CombinationIterator(string s, int len) {
        v.clear();
        int l = s.size();
        
        for ( int mask = 1; mask < ( 1<<l ); mask++ ) {
            string ret;
            int cnt = 0;
            for ( int i = 0; i < l; i++ ) {
                if ( ( 1 << i ) & mask ) {
                    ret += s[i];
                    cnt++;
                }
            }
            
            if ( cnt == len ) {
                v.push_back ( ret );
            }
        }
        
        sort ( v.begin(), v.end() );
        reverse ( v.begin(), v.end() );
    }
    
    string next() {
        string ret = v.back();
        v.pop_back();
        return ret;
    }
    
    bool hasNext() {
        return !v.empty();   
    }
};

/**
 * Your CombinationIterator object will be instantiated and called as such:
 * CombinationIterator* obj = new CombinationIterator(characters, combinationLength);
 * string param_1 = obj->next();
 * bool param_2 = obj->hasNext();
 */